core/src/ldml/ldml_processor.cpp

```diff
@@ -330,14 +337,18 @@ size_t ldml_processor::process_output(ldml_event_state &ldml_state, const std::u
 
   // drop last 'matchedContext':
   ctxtstr.resize(ctxtstr.length() - matchedContext);
-  ctxtstr.append(outputString); // TODO-LDML: should be able to do a normalization-safe append here.
-  (void)ldml::normalize_nfd_markers(ctxtstr);
-
+  ctxtstr.append(outputString); // TODO-LDML: should be able to do a normalization-safe append here.'
+  if (!normalization_disabled) {
+    (void)ldml::normalize_nfd_markers(ctxtstr);
```

Comment:
Why no assertion on this instance? Seems inconsistent (See extracting-it-out comment above)

Reply:
@mcdurdin perhaps an inline member would make sense.